connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java

```diff
@@ -60,17 +60,13 @@
 
     private static final String TASK_MAX_DISPLAY = "Tasks max";
 
-    protected static ConfigDef config;
-
-    static {
-        config = new ConfigDef()
-                .define(NAME_CONFIG, Type.STRING, Importance.HIGH, NAME_DOC, COMMON_GROUP, 1, Width.MEDIUM, NAME_DISPLAY)
-                .define(CONNECTOR_CLASS_CONFIG, Type.STRING, Importance.HIGH, CONNECTOR_CLASS_DOC, COMMON_GROUP, 2, Width.LONG, CONNECTOR_CLASS_DISPLAY)
-                .define(TASKS_MAX_CONFIG, Type.INT, TASKS_MAX_DEFAULT,  atLeast(TASKS_MIN_CONFIG), Importance.HIGH, TASKS_MAX_DOC, COMMON_GROUP, 3, Width.SHORT, TASK_MAX_DISPLAY);
-    }
+    private static ConfigDef config = configDef();
```

Comment:
I don't think we want this in the base class -- its risky as it leads to accidental sharing as we had before. Instantiating it once statically in `SourceConnectorConfig` and `SinkConnectorConfig` seems fine, but keeping it here seems like we're inviting people to accidentally define additional parameters on it.

It looks like we only use this in one place (in this file) so probably not a big deal to remove it.